### DISCUSSION TITLE
Asset URL behind proxy server

### DISCUSSION THREAD

**Discussion by nilskretschmer** [laravel/framework#48793] [Q&A]
I deployed my Laravel App to an enterprise environment in production. The app and all related services are running as docker services. Because of different networking segments the app can be reached internally and externally using different DNS names. For this I use nginx as a proxy server which proxy-passes external DNS requests to the internal Laravel app.

Now I have the following problem:
1. The Laravel App is only running on the internal network (1 service)
2. The app needs to be reached with multiple DNS names (one internal, two external)
3. For the external access I use separate nginx instances as proxy servers (one for each external DNS/network segment)
4. If I access the app I get a status code 200 for the page request itself using the external DNS name
5. Now the browser makes GET requests to get the app's assets (like .css, .js files built by vite)
6. The browser tries to get those assets using the internal DNS name which can't be resolved externally and so I just get a blank page

How can I tell Laravel that asset's should be served using the requesting hostname? I can't just set the ASSET_URL environment variable, because the DNS name is different depending on the network I make the request from. So the asset url is dynamic not static.

**Comment by nilskretschmer** [comment]
For anyone wondering how to solve this:
It seems like Laravel is also using the X-Forwarded-Host Header for the asset urls if it is present in the request. In case of (multiple) proxy servers just make sure that the header is passed through every proxy to the Laravel App. Then all Asset URLs will use the domain from the X-Forwarded-Host.

In my case the second proxy was overwriting the X-Forwarded-Host from the first proxy which led to the described problems.

**Comment by raviprakash007** [comment]
In Proxy setting use $http_host , in place of $host.

```
     proxy_set_header` Host $http_host;
     proxy_set_header X-Forwarded-Host $http_host;
```
